### PR TITLE
Interactive certificate selection with install verb

### PR DESCRIPTION
**Pull request by joohoi**
If either `--cert-name` or both `--key-path` and `--cert-path` (in which case the user requests installation for a certificate not managed by Certbot) are not provided, prompt the user with managed certificates and let them choose.

Fixes: #5824

### CODE REVIEW

File: certbot/main.py [hunk 1]
```diff
@@ -790,6 +790,13 @@ def install(config, plugins):
     except errors.PluginSelectionError as e:
         return str(e)
 
+    custom_cert = (config.key_path and config.cert_path)
```
Comment (ohemorange) [inline certbot/main.py]: Should also change line 805 to match

Comment (joohoi) [reply certbot/main.py]: The current implementation on line 805 catches both of the cases - user providing both `--key-path` and `--cert-path` and configuration object values being populated from either interactively selected or user provided `--cert-name`.

Are you suggesting something in line with `if custom_cert or config.certname:` ?

Comment (ohemorange) [reply certbot/main.py]: I meant this as a nit -- you already create a variable that captures this boolean, so might as well use `custom_cert` on line 805 in place of `config.key_path and config.cert_path`

Comment (joohoi) [reply certbot/main.py]: Ah, right. Maybe I should elaborate a bit; these values get populated from `config.certname` obtained from CLI or by interactive selection in line 804 to normalize the handling from there on, so the initial variable value might not reflect the actual state at this point.

**Comment by ohemorange** [comment]
I'll let you merge because of the enhancements stuff that's now in there

**Comment by joohoi** [comment]
Ready for the next review round, I had to make the config mock more detailed in two of the `main_test.py` tests.

**Pull request merged**